src/Lean/Meta/Basic.lean

```diff
@@ -1347,6 +1347,16 @@ private def withNewMCtxDepthImp (allowLevelAssignments : Bool) (x : MetaM α) :
   finally
     modify fun s => { s with mctx := saved.mctx, postponed := saved.postponed }
 
+/--
+Removes `fvarId` from the local context, and replaces occurrences of it with `e`.
+It is the responsibility of the caller to ensure that `e` is well-typed in the context
+of any occurrence of `fvarId`.
```

Comment:
This operation sounds like "substitution" to me, but you're not using that terminology. Why not?

Reply:
No deep reason, just out of consistency with existing function names and doc strings: https://leanprover-community.github.io/mathlib4_docs/Lean/Expr.html#Lean.Expr.replaceFVar